File: frontend/src/types/expense.ts
export type ExpensePaymentMethod =
  | "Cash"
  | "MTN_MOMO"
  | "ORANGE_MONEY"
  | "Card"
  | "BankTransfer";

export interface Expense {
  id?: string;
  expenseNumber?: string;
  category: string;
  description: string;
  amount: number;
  paymentMethod: ExpensePaymentMethod;
  referenceNumber: string;
  notes: string;
  status?: string;
  expenseDate: string;
  createdAt?: string;
  updatedAt?: string;
}

File: backend/BarMaster.API/Models/Expense.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace BarMaster.API.Models;

public class Expense
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Id { get; set; }

    [BsonElement("expenseNumber")]
    public string ExpenseNumber { get; set; } = string.Empty;

    [BsonElement("category")]
    public string Category { get; set; } = string.Empty;

    [BsonElement("description")]
    public string Description { get; set; } = string.Empty;

    [BsonElement("amount")]
    public decimal Amount { get; set; }

    [BsonElement("paymentMethod")]
    public string PaymentMethod { get; set; } = "Cash";

    [BsonElement("referenceNumber")]
    public string ReferenceNumber { get; set; } = string.Empty;

    [BsonElement("notes")]
    public string Notes { get; set; } = string.Empty;

    [BsonElement("status")]
    public string Status { get; set; } = "Completed";

    [BsonElement("expenseDate")]
    public DateTime ExpenseDate { get; set; } = DateTime.UtcNow;

    [BsonElement("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [BsonElement("updatedAt")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}